app/src/lib/git/commit.ts

```diff
@@ -22,23 +29,57 @@ export async function createCommit(
     })
     return true
   } catch (e) {
-    // Commit failures could come from a pre-commit hook rejection. So display
-    // a bit more context than we otherwise would.
-    if (e instanceof GitError) {
-      const output = e.result.stderr.trim()
-
-      let standardError = ''
-      if (output.length > 0) {
-        standardError = `, with output: '${output}'`
-      }
-      const exitCode = e.result.exitCode
-      const error = new Error(
-        `Commit failed - exit code ${exitCode} received${standardError}`
-      )
-      error.name = 'commit-failed'
-      throw error
-    } else {
-      throw e
+    handleCommitError(e)
+    return false
+  }
+}
+
+/**
+ * Creates a commit to finish an in-progress merge
+ * assumes that all conflicts have already been resolved
+ *
+ * @param repository repository to execute merge in
+ * @param files files to commit
+ */
+export async function createMergeCommit(
```

Comment:
It'd be great to get some tests in for this new function while you're sketching out the API (`createCommit` already has some decent coverage).

Ideas that come to mind:
 - happy path testing - repository in conflict, commit files with conflicts, does it succeed? does a merge commit get created?
 - unresolved conflicts - I suspect unstaging all files as part of the operation prevents from occurring, but maybe there's a way for Git to interrupt you?
 - creating a merge commit when not in conflicted state - return false, or throw error?

For higher-level concerns (like ensuring conflict markers are removed) I wouldn't worry about them here, and just let the user proceed.

Reply:
yes, agreed! :) i'm already in the process of adding some tests. will update when its ready.